### ISSUE TITLE
Running scoring in notebooks results in a traceback 

### ISSUE THREAD

**Issue by Sumukh** [Cal-CS-61A-Staff/ok-client#298]
Regression. Worked fine in 1.10.7

```
# Ok Autograding Start
_ = ok.score(score_out='score-file.txt')
# Ok Autograding Run

```

    ---------------------------------------------------------------------------

    KeyError                                  Traceback (most recent call last)

    <ipython-input-47-30dd4d9dcab5> in <module>()
          1 # Ok Autograding Start
    ----> 2 _ = ok.score(score_out='score-file.txt')
          3 # Ok Autograding Run

    /usr/local/lib/python3.5/dist-packages/client/api/notebook.py in score(self, env, score_out)
         48             env = __main__.__dict__
         49         self.run('scoring', messages, env=env)
    ---> 50         return messages['scoring']
         51 
         52     def backup(self):

    KeyError: 'scoring'
```

~It seems to work sometimes though...~ (Edit: I think that's because it was routed a node with image that had an older version of ok-client - this should be reproducible)

**Comment by Sumukh** [comment]
It'd be good to fix this - but the workaround for now is that the autograding image just uses `1.10.7`.

**Comment by epai** [comment]
This was fixed in `1.12.5`.  Problem was `assignment.set_args` wasn't mutating the settings object, so protocols weren't getting the updated args.
